toolchain/check/check.cpp

```diff
@@ -200,6 +200,81 @@ struct UnitInfo {
   UnitInfo* api_for_impl = nullptr;
 };
 
+// Imports the current package.
+static auto ImportCurrentPackage(Context& context, UnitInfo& unit_info,
+                                 int total_ir_count,
+                                 SemIR::InstId package_inst_id,
+                                 SemIR::TypeId namespace_type_id) -> void {
+  // Add imports from the current package.
+  auto self_import = unit_info.package_imports_map.find(IdentifierId::Invalid);
+  if (self_import == unit_info.package_imports_map.end()) {
+    // Push the scope; there are no names to add.
+    context.scope_stack().Push(package_inst_id, SemIR::NameScopeId::Package);
+    return;
+  }
+
+  auto imported_irs = llvm::SmallVector<bool>(total_ir_count, false);
```

Comment:
Yeah, a single allocation once per invocation seems kinda irrelevant. I think that keeping the reference / re-indexing is also irrelevant (and I would expect there to actually be zero cost to re-indexing -- the optimizer should see right through that) for the same reason. So maybe `SmallVector` is better just to avoid using a less-well-known type.